### DISCUSSION TITLE
A proposal to use Better Auth instead of Next Auth as a the auth provider

### DISCUSSION THREAD

**Discussion by Bekacru** [t3-oss/create-t3-turbo#1241] [Ideas]
### Describe the feature you'd like to request

The current implementation of Auth requires a lot of workarounds in the user land to make NextAuth work for both Expo and nextjs, as well as to enable preview deployments.

### Describe the solution you'd like to see

[Better Auth](https://better-auth.com) is a new authentication library with a plugin ecosystem, allowing for easier integration of features like these. It includes Expo integration that works out of the box and supports OAuth logins for preview deployments without needing a proxy server.

### Additional information

Perhaps before this is considered, Better Auth should be an optional authentication provider in the Create T3 App. I already have a fork that implements Better Auth for Create T3 Turbo, which you can take a look at.

https://github.com/Bekacru/create-t3-turbo

**Comment by nktnet1** [comment]
Hey @Bekacru,

When I tried 
```sh
npx create-turbo@latest -e https://github.com/Bekacru/create-t3-turbo
```

I did get the error:
```text
│ > pnpm dlx sherif@latest
│ 1 issue found in ./:
│  ⨯ error Dependency @better-auth/expo has multiple versions defined in the workspace. multiple-dependency-versions
│   ./apps
│       expo                      ^1.1.15-beta.7   ↑ highest
│   ./packages
│       auth                      1.1.15-beta.7   ↓ lowest
│ 1 issue found (1 ⨯, 0 ⚠️, 0 ✓) across 12 packages in 542.228µs.
│  Note: use `-i` to ignore dependencies, `-r` to ignore rules, `-p` to ignore packages, and `-f` to autofix fixable issues.
│  ELIFECYCLE  Command failed with exit code 1.
```

and had to manually change `./packages/auth/package.json` to `^1.1.15-beta.7` before reinstalling.

Would you be able to fix this, as well as consider:
- enabling issues & discussions in the fork
- syncing the latest commits from upstream

Thank you!

**Reply by Hidde-Heijnen** [reply]
still having issues even after fixing this! unfortunately

**Reply by nktnet1** [reply]
@Hidde-Heijnen yeah, there's probably a few other dependencies that you will need to align for it to work.

I made a turborepo for a different stack (tanstack router + hono instead of nextjs), but I was able to integrate it successfully with Better-Auth - you can copy the relevant parts from here or use it as reference if you need:
- https://github.com/nktnet1/rt-stack

**Reply by kodermax** [reply]
Can someone do more with bunjs?

**Comment by inducingchaos** [comment]
Came here while migrating! The amount of code I can delete with Better Auth is insane. Great work.
